### PR TITLE
Add read/write PubSub integration example fhirio pipeline

### LINKED ISSUE #22305: [Task]: Add sample fhirio pipeline
### What needs to happen?

Add sample Go fhirio pipeline including pubsub integration.

### Issue Priority

Priority: 2

### Issue Component

Component: io-go-gcp

### PR DESCRIPTION
**Pull request by lnogueir**
Add a sample `fhirio` pipeline to demonstrate usage of `Read` and `ExecuteBundles` transforms as well as integration with `pubsubio`.

Fixes #22305

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by lnogueir** [comment]
R: @msbukal

File: sdks/go/pkg/beam/io/fhirio/common.go [hunk 1]
```diff
@@ -73,8 +74,8 @@ type operationResults struct {
 }
 
 type fhirStoreClient interface {
-	readResource(resourcePath string) (*http.Response, error)
-	executeBundle(storePath string, bundle []byte) (*http.Response, error)
+	readResource(resourcePath []byte) (*http.Response, error)
+	executeBundle(storePath string, bundle string) (*http.Response, error)
```
Comment (msbukal) [inline sdks/go/pkg/beam/io/fhirio/common.go:78]: Why the swap here?

**Review by msbukal — APPROVED** [review]

**Comment by lnogueir** [comment]
Run GoPortable PreCommit

**Comment by lostluck** [comment]
I'm currently busy with a conference in Berlin, but the rest of my team should be helpful!

**Comment by lostluck** [comment]
assign to next reviewer

**Review by damccorm — APPROVED** [review]
LGTM - thanks!

**Pull request merged**